src/test_typing.py

```diff
@@ -2057,8 +2051,14 @@ class NonDefaultAfterDefault(NamedTuple):
     def test_annotation_usage_with_methods(self):
         self.assertEquals(XMeth(1).double(), 2)
         self.assertEquals(XMeth(42).x, XMeth(42)[0])
-        self.assertEquals(XMethBad(1)._fields, ('x',))
-        self.assertEquals(XMethBad(1).__annotations__, {'x': int})
+
+        with self.assertRaises(AttributeError):
```

Comment:
Thanks you for suggestion. I actually tried this and I now realise that it is not possible to raise an error in this case - defining a method with the same name as a field is interpreted as a default value. But I think this is OK, currently these are all equivalent at runtime:
```python
class A(NamedTuple):
    a: Callable = lambda s: s
class A(NamedTuple):
    a: Callable
    a = lambda s: s
class A(NamedTuple):
    a: Callable
    def a(s):
        return s
```
Some of those could be prohibited in mypy however.